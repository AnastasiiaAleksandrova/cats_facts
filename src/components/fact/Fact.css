.fact {
    padding: .5em;
    margin: .5em;
}

/*.fact:hover {
    transform: scale(1.1);
    cursor: pointer;
}*/