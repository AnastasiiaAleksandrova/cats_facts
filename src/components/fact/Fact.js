import React from "react";
import { Paper } from "@material-ui/core";
import "./Fact.css";


function Fact(props) {
    return (
        <Paper className="fact">
            {props.text}
        </Paper>
    );
}

export default Fact;
