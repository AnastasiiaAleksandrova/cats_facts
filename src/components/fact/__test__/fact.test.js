import React from "react";
import Fact from "../Fact";
import { render, unmountComponentAtNode } from "react-dom";
import renderer from 'react-test-renderer';

let container = null;

beforeEach(() => {
    container = document.createElement("div");
    document.body.appendChild(container);
});

afterEach(() => {
    unmountComponentAtNode(container);
    container.remove();
    container = null;
});

it("renders without crashing and contains a fact ", () => {
    render(<Fact text="This is a cat fact" />, container);
    expect(container.textContent).toBe("This is a cat fact");
});

it("matches snapshot", () => {
    const component = renderer.create(<Fact />).toJSON();
    expect(component).toMatchSnapshot();
})