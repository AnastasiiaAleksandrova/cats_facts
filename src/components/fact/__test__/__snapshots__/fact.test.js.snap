// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="MuiPaper-root fact MuiPaper-elevation1 MuiPaper-rounded"
/>
`;
